Automattic/wp-calypso
# E2E: update Site Assembler flow to support the multi domain selection flow.

Related to https://github.com/Automattic/dotcom-forge/issues/4371, https://github.com/Automattic/dotcom-forge/issues/4371.

## Proposed Changes

This PR updates the Site Assembler E2E spec to support the multi-domain selection flow.

Additionally, there are some updates and refactor to the methods involved in the flow.

## Testing Instructions

Ensure the following build configurations are passing:

Triggered a run on CI, all passed (minus one flaky test that passed on second try):
![image](https://github.com/Automattic/wp-calypso/assets/6549265/8a532b57-2390-4597-95b9-83643358dedf)


## Pre-merge Checklist
